src/S3/MultipartCopy.php

```diff
@@ -107,19 +112,27 @@ protected function getUploadCommands(callable $resultHandler)
 
     private function createPart($partNumber, $partsCount)
     {
+        $data = [];
+
+        // Apply custom params to UploadPartCopy data
+        $config = $this->getConfig();
+        $params = isset($config['params']) ? $config['params'] : [];
+        foreach ($params as $k => $v) {
+            $data[$k] = $v;
```

Comment:
Any granularity on customizing these operations should still be handled through the 'before_X' callbacks, including the new `before_lookup` in ObjectCopier. The changes around 'params' here are to build out handling the contracts in the ObjectUploader/ObjectCopier and also allow ease of use for the parameters that are consistent across the operations.